infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/LogReplicationEvent.java

```diff
@@ -51,13 +52,16 @@ public enum LogReplicationEventType {
      */
     private LogReplicationEventMetadata metadata;
 
+    @Getter
+    private LogReplicationFSM replicationFsm;
+
     /**
      * Constructor
      *
      * @param type log replication event type
      */
-    public LogReplicationEvent(LogReplicationEventType type) {
-        this(type, LogReplicationEventMetadata.empty());
+    public LogReplicationEvent(LogReplicationEventType type, LogReplicationFSM replicationFsm) {
+        this(type, LogReplicationEventMetadata.empty(), replicationFsm);
```

Comment:
Yes, its needed for the taskManager to identify which fsm an event belongs too....needed for where there are multiple sessions.
Since taskManager is shared by all the sessions, LogReplicationFSM cannot be part of taskManager's constructor. The relation is : which FSM does the current event belong to.